### PR TITLE
openixcard: new, 1.0.1

### PR DESCRIPTION
**Pull request by CamberLoid**
Topic Description
-----------------

The topic will introduce a new package `openixcard`, which is helpful for developing BSP for sunxi target.

Package(s) Affected
-------------------

`openixcard`: 1.0.1

Security Update?
----------------

No.

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
    

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] RISC-V 64-bit `riscv64`

### CODE REVIEW

**Comment by CamberLoid** [comment]
LGTM on AMD64. May not test on other platform.
(Wrong image error corrected)
image
image

**Comment by CamberLoid** [comment]
For the record: patch is related to https://github.com/YuzukiTsuru/OpenixCard/issues/3

**Comment by MingcongBai** [comment]
> For the record: patch is related to [YuzukiTsuru/OpenixCard#3](https://github.com/YuzukiTsuru/OpenixCard/issues/3)

You should include this in your commit msg, as a "Ref: " line.

File: extra-devel/openixcard/autobuild/defines [hunk 1]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=openixcard
+PKGSEC=devel
+PKGDES="Open Source Implementation of Allwinner/sunxi's Image Manipulation Tool PhoenixCard"
+
+BUILDDEP="cmake"
```
Comment (MingcongBai) [inline extra-devel/openixcard/autobuild/defines]: You won't need to define this. Our guideline specifies that there is no need to specify software shipped with BuildKit as BUILDDEP.

File: extra-devel/openixcard/autobuild/defines [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+PKGDEP="glibc confuse"
+
+ABTYPE=cmake
```
Comment (MingcongBai) [inline extra-devel/openixcard/autobuild/defines]: Would `cmakeninja` not work?

File: extra-devel/openixcard/autobuild/beyond [hunk 3]
```diff
@@ -0,0 +1,2 @@
+abinfo "Fixing wrong permissions ..."
+chmod +x "$PKGDIR"/usr/libexec/openixcard/genimage
```
Comment (MingcongBai) [inline extra-devel/openixcard/autobuild/beyond]: Missing verbosity.

File: extra-devel/openixcard/autobuild/prepare [hunk 4]
```diff
@@ -0,0 +1,4 @@
+abinfo "Patching submodule lib/argparse ..."
+pushd lib/argparse
```
Comment (MingcongBai) [inline extra-devel/openixcard/autobuild/prepare]: Use absolute path here.

File: extra-devel/openixcard/autobuild/defines [hunk 5]
```diff
@@ -0,0 +1,8 @@
+PKGNAME=openixcard
+PKGSEC=devel
+PKGDES="Open Source Implementation of Allwinner/sunxi's Image Manipulation Tool PhoenixCard"
```
Comment (MingcongBai) [inline extra-devel/openixcard/autobuild/defines]: "Open source implementation of PhoenixCard - Allwinner's image manipulation tool"

Reader attention theory: keep the key information/detail towards the middle of the sentence.

Also, there's no need to capitalise all first letters for all words in a sentence.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**